src/Compile.hs

```diff
@@ -106,12 +107,13 @@ instance Compilable Exp where
         <> ")"
      where
       compilePattern :: String -> Pattern -> String
-      compilePattern _     (PVar _) = "true"
-      compilePattern _     PAny     = "true"
-      compilePattern scope (PNum n) = scope <> " === " <> n
+      compilePattern _     (PVar _)                = "true"
+      compilePattern _     PAny                    = "true"
+      compilePattern scope (PNum n)                = scope <> " === " <> n
       compilePattern scope (PStr n) = scope <> " === \"" <> n <> "\""
-      compilePattern scope (PBool n) | n == "True" = scope <> " === true"
-                                     | otherwise   = scope <> " === false"
+      compilePattern scope (PBool n) | n == "true" = scope <> " === true"
+      compilePattern scope (PBool n) | True      = scope <> " === true"
```

Comment:
```suggestion
      compilePattern scope (PBool n)  = scope <> " === " <> n
```

Reply:
`otherwise` is actually True under the hood. Therefore here I'm pretty sure that it would always resolve to `true` and even though it'd compile because it's still a Bool and typecheck works, it'd compile the wrong thing.

Reply:
Ahhh it's for the patterns, we might not have a pattern with a falsy bool, but nevertheless my proposed change should be correct.

Reply:
Ok, line 34:
```
LBool v     -> toLower <$> v
```

We can remove the toLower here, and just use my proposed change where appropriate ( line 114 also )

Reply:
So that's why it mostly works still.